src/datamigration/azext_datamigration/manual/helper.py

```diff
@@ -460,8 +500,35 @@ def is_user_admin():
 # Helper function to validate key input.
 # -----------------------------------------------------------------------------------------------------------------
 def validate_input(key):
-    if key == "":
-        raise InvalidArgumentValueError("Failed: IR Auth key is empty. Please provide a valid auth key.")
+
+    if is_string_null_or_empty(key):
+        raise InvalidArgumentValueError("Failed: IR Auth key is null or empty. Please provide a valid auth key.")
+    if has_shell_escape_characters(key):
+        raise InvalidArgumentValueError("Failed: IR Auth key has invalid characters. Please provide a valid auth key.")
+    if not is_valid_ir_key_format(key):
+        raise InvalidArgumentValueError("Failed: Invalid IR Auth key format. Please provide a valid auth key.")
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check if the input key is in valid format.
+# -----------------------------------------------------------------------------------------------------------------
+def is_valid_ir_key_format(key):
+
+    keyPrefix = "IR"
+    separator = '@'
+
+    key_parts = key.split(separator)
+
+    # Length of the split auth key should be 5
+    if len(key_parts) != 5:
```

Comment:
Make this `5` as a constant. Magic numbers are not good.

Reply:
Done